pandas/_libs/window.pyx

```diff
@@ -1391,7 +1392,17 @@ def roll_quantile(ndarray[float64_t, cast=True] input, int64_t win,
 
         if nobs >= minp:
             idx = int(quantile * <double>(nobs - 1))
-            output[i] = skiplist.get(idx)
+
+            # Single value in skip list
+            if nobs == 1:
+                output[i] = skiplist.get(0)
+
+            # Interpolated quantile
+            else:
+                vlow = skiplist.get(idx)
+                vhigh = skiplist.get(idx + 1)
```

Comment:
Hadn't realized this before: `skiplist.get` is an O(log win) operation which we now have to perform twice. We also do a `skiplist.insert` and a `skiplist.remove`, which are also O(log win), and probably more expensive than `skiplist.get`, but a pessimistic estimation says the performance of this function will go down by 25%.

I think there are two things to do here:

 1. quantify that slow down by running a couple of benchmarks with and without this PR, and leave a record of the results for further reference.
 2. if the slowdown is significant, we could gain the speed back by doing some changes to the skiplist internals: when the first index is fetched, getting the next one is trivially simple and cheap, so we could add a `.get_two()` method to `IndexableSkiplist` and use it here. I think that would be work for another PR, but creating an issue for tracking based on the results of the benchmarks should be part of merging this.